plugins/cros-ec/fu-cros-ec-usb-device.c

```diff
@@ -812,18 +902,29 @@ fu_cros_ec_usb_device_attach (FuDevice *device, GError **error)
 }
 
 static gboolean
-fu_cros_ec_usb_device_detach (FuDevice *self, GError **error)
+fu_cros_ec_usb_device_detach (FuDevice *device, GError **error)
 {
-	if (fu_device_has_custom_flag (self, "rw-written") &&
-	    !fu_device_has_custom_flag (self, "ro-written"))
+	FuCrosEcUsbDevice *self = FU_CROS_EC_USB_DEVICE (device);
+
+	if (fu_device_has_custom_flag (device, "rw-written") &&
+	    !fu_device_has_custom_flag (device, "ro-written"))
 		return TRUE;
 
-	fu_device_set_remove_delay (self, FU_DEVICE_REMOVE_DELAY_RE_ENUMERATE);
-	if (!fu_cros_ec_usb_device_reset_to_ro (self, error))
-		return FALSE;
+	if (self->in_bootloader) {
+		g_print ("skipping immediate reboot in case of already in bootloader");
+		/* in RO so skip reboot */
+		return TRUE;
+	} else if (self->targ.common.flash_protection != 0x0) {
+		/* in RW, and RO region is write protected, so jump to RO */
+		fu_device_set_custom_flags (device, "ro-written");
+		fu_device_set_remove_delay (device, CROS_EC_REMOVE_DELAY_RE_ENUMERATE);
+		if (!fu_cros_ec_usb_device_reset_to_ro (device, error)) {
+			return FALSE;
+		}
```

Comment:
braces not required here, but otherwise generally looks good; Mario's comments are all valid.

Reply:
Done.